crates/uv-python/src/discovery.rs

```diff
@@ -944,11 +945,25 @@ pub(crate) fn find_python_installation(
             continue;
         }
 
+        // If it's an alternative implementation, and alternative implementations aren't allowed
+        // skip it; we avoid querying these interpreters at all if they're on the search path and
+        // are not requested, but other sources such as the managed installations will include them.
+        if installation.is_alternative_implementation()
+            && !request.allows_alternative_implementations()
+            && !installation.source.allows_alternative_implementations()
+            // Installing an alternative implementation with a default executable name is opt-in
+            && !(installation.interpreter.has_default_executable_name()
+                && installation.source == PythonSource::SearchPath)
```

Comment:
I think I need this part explained. This is, like, if the executable is named `python` but it's really PyPy?

Reply:
I futzed with the commentary and implementation here a fair bit. I would kind of like to track the executable name on `PythonSource::SearchPath` and implement this in `PythonSource::allows_alternative_implementations()` but that would be fairly disruptive.